### PR TITLE
Ensure HDD1 is never damaged or corrupted due to unexpected RPCS3/game close

### LINKED ISSUE #14958: The Last Of Us won't boot normally ever after an abrupt shutdown during first load screen unless HDD1 is cleaned
The reason is clear, this is caused by partial data in HDD1. Unless a process exists explicitly with sys_process_exit all HDD1 data should be treated as damaged/incomplete and be cleared on the next boot. This is probably what PS3 does as well. Noting for myself to fix this later.

### PR DESCRIPTION
**Pull request by elad335**
Fixes #14958

### CODE REVIEW

File: rpcs3/Emu/VFS.cpp [hunk 1]
```diff
@@ -1056,17 +1066,35 @@ bool vfs::host::unlink(const std::string& path, [[maybe_unused]] const std::stri
 #endif
 }
 
-bool vfs::host::remove_all(const std::string& path, [[maybe_unused]] const std::string& dev_root, [[maybe_unused]] const lv2_fs_mount_point* mp, bool remove_root, [[maybe_unused]] bool lock)
+bool vfs::host::remove_all(const std::string& path, [[maybe_unused]] const std::string& dev_root, [[maybe_unused]] const lv2_fs_mount_point* mp, bool remove_root, [[maybe_unused]] bool lock, bool force_atomic)
 {
-#ifdef _WIN32
+	static_cast<void>(force_atomic);
```
Comment (Megamouse) [inline rpcs3/Emu/VFS.cpp:1071]: Use maybe_unused instead

File: rpcs3/Emu/Cell/Modules/cellSysCache.cpp [hunk 2]
```diff
@@ -31,6 +31,19 @@ void fmt_class_string<CellSysCacheError>::format(std::string& out, u64 arg)
 
 extern lv2_fs_mount_point g_mp_sys_dev_hdd1;
 
+extern std::string get_syscache_state_corruption_indicator_file_path(std::string_view dir_path)
+{
+	constexpr std::stringu8_view append_path = u8"/＄hdd0_temp_state_indicator";
+	const auto filename = reinterpret_cast<const char*>(append_path.data());
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/Modules/cellSysCache.cpp:37]: Unused

File: rpcs3/Emu/Cell/Modules/cellSysCache.cpp [hunk 3]
```diff
@@ -141,8 +193,8 @@ error_code cellSysCacheMount(vm::ptr<CellSysCacheParam> param)
 
 	std::lock_guard lock0(g_mp_sys_dev_hdd1.mutex);
 
-	// Check if can reuse existing cache (won't if cache id is an empty string)
-	if (param->cacheId[0] && cache_id == cache.cache_id)
+	// Check if can reuse existing cache (won't if cache id is an empty string or cache is damaned/incomplete)
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/Modules/cellSysCache.cpp:196]:
```suggestion
	// Check if can reuse existing cache (won't if cache id is an empty string or cache is damaged/incomplete)
```

File: rpcs3/Emu/Memory/vm_ptr.h [hunk 4]
```diff
@@ -452,3 +452,15 @@ struct fmt_class_string<vm::_ptr_base<char, u32>, void> : fmt_class_string<vm::_
...
+struct fmt_class_string<vm::_ptr_base<const char[Size], u32>, void> : fmt_class_string<vm::_ptr_base<const char, u32>>
+{
+	// Classify const char[] as const char*
+};
+
+template <usz Size>
+struct fmt_class_string<vm::_ptr_base<char[Size], u32>, void> : fmt_class_string<vm::_ptr_base<const char, u32>>
+{
+	// Classify char[] as const char*
+};
```
Comment (Megamouse) [inline rpcs3/Emu/Memory/vm_ptr.h:466]:
```suggestion
};

```

**Comment by ChwaraeTeg** [comment]
Just to confirm, this doesn't fix #15017 - the issue with that one seems to relate to HDD0 rather than HDD1

**Pull request merged**